### ISSUE TITLE
AttributeError: 'module' object has no attribute 'alpha_composite'

### ISSUE THREAD

**Issue by renickbell** [fablabnbg/inkscape-centerline-trace#26]
Any idea about this? Version 0.7a, Inkscape 0.92.2 2405546, 2018-03-11

Traceback (most recent call last):
  File "centerline-trace.py", line 454, in <module>
    e.affect()
  File "/usr/share/inkscape/extensions/inkex.py", line 283, in affect
    self.effect()
  File "centerline-trace.py", line 415, in effect
    path_svg,stroke_width,im_size = self.svg_centerline_trace(filename)
  File "centerline-trace.py", line 193, in svg_centerline_trace
    im = Image.alpha_composite(bg, im)
AttributeError: 'module' object has no attribute 'alpha_composite'

**Comment by jnweiger** [comment]
Image.alpha_composite() is from PIL. Can you tell, which version of PIL you are using.
I have this: ```

grep VERSION /usr/lib/python3/dist-packages/PIL/__init__.py 
VERSION = '1.1.7'  # PIL Version

**Comment by renickbell** [comment]
according to:
https://aur.archlinux.org/packages/python2-imaging/
pil=1.1.7

**Comment by jnweiger** [comment]
Hmm, so maybe the code is plain wrong. I cannot test that at all currently, as I don't have a working autotrace here.

**Comment by jnweiger** [comment]
Can you reproduce this?
```
python
Python 2.7.15rc1 (default, Apr 15 2018, 21:51:34) 
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from PIL import Image
>>> Image.alpha_composite
<function alpha_composite at 0x7f7b8a721de8>
```

**Comment by Moini** [comment]
For comparison: I'm not getting the error with 0.7a on Linux Mint with Inkscape 0.92.x (self-compiled, should be irrelevant). Just in case, I also tested with a black-stroke-on-transparent-bg image. I can import the module just fine.

**Comment by jnweiger** [comment]
@renickbell does it crash for you every time, or only for one specific image?

**Comment by renickbell** [comment]
Solved!

OK. Thanks for your help. Arch Linux had two libraries: one providing PIL (which gives the results to your question above and my error).

I uninstalled it and installed the python2 version of pillow, which then allowed me to successful execute in the terminal what you showed above. It also allows the plugin to work in Inkscape.

I'll try to report this to the package maintainer for the python2-imaging package.

python2.7
Python 2.7.14 (default, Jan  5 2018, 10:41:29)
[GCC 7.2.1 20171224] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> ;2~
KeyboardInterrupt
>>> from PIL import Image
>>> Image.alpha_composite
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'alpha_composite'
